SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriverGameplayDebuggerContext.cpp

```diff
@@ -0,0 +1,422 @@
+		return;
+	}
+
+	LBStrategy = NewObject<UGameplayDebuggerLBStrategy>();
+	LBStrategy->Init(*this, *NetDriver->LoadBalanceStrategy);
+	NetDriver->LoadBalanceStrategy = LBStrategy;
+	NetDriver->Sender->UpdatePartitionEntityInterestAndPosition();
+
+	TSet<VirtualWorkerId> VirtualWorkerIds = LBStrategy->GetVirtualWorkerIds();
+
+	PhysicalToVirtualWorkerIdMap.Reserve(VirtualWorkerIds.Num());
+	for (const auto& VirtualWorkerId : VirtualWorkerIds)
+	{
+		const FString* PhysicalWorkerName = NetDriver->VirtualWorkerTranslator->GetPhysicalWorkerForVirtualWorker(VirtualWorkerId);
+		if (PhysicalWorkerName != nullptr)
+		{
+			PhysicalToVirtualWorkerIdMap.Add(*PhysicalWorkerName, VirtualWorkerId);
+		}
+		else
+		{
+			UE_LOG(LogSpatialNetDriverGameplayDebuggerContext, Error, TEXT("Failed to convert virtual worker to physical worker name"));
+		}
+	}
+}
+
+void USpatialNetDriverGameplayDebuggerContext::Cleanup()
+{
+	check(NetDriver && NetDriver->Sender);
+
+	Reset();
+
+	if (LBStrategy != nullptr)
+	{
+		UAbstractLBStrategy* WrappedStrategy = LBStrategy->GetWrappedStrategy();
+		if (WrappedStrategy != nullptr)
+		{
+			NetDriver->LoadBalanceStrategy = LBStrategy->GetWrappedStrategy();
+			NetDriver->GameplayDebuggerCtx = nullptr;
+			NetDriver->Sender->UpdatePartitionEntityInterestAndPosition();
+		}
+		else
+		{
+			UE_LOG(LogSpatialNetDriverGameplayDebuggerContext, Error, TEXT("Cleanup without a wrapped strategy"));
+		}
+	}
+}
+
+void USpatialNetDriverGameplayDebuggerContext::Reset()
+{
+	check(NetDriver && NetDriver->Connection && NetDriver->Sender);
+
+	for (const auto& Entry : NetDriver->Connection->GetView())
+	{
+		const SpatialGDK::EntityViewElement& ViewElement = Entry.Value;
+		if (ViewElement.Authority.Contains(SpatialConstants::SERVER_AUTH_COMPONENT_SET_ID)
+			&& ViewElement.Components.ContainsByPredicate([](const SpatialGDK::ComponentData& Data) {
+				   return Data.GetComponentId() == SpatialConstants::GDK_GAMEPLAY_DEBUGGER_COMPONENT_ID;
+			   }))
+		{
+			NetDriver->Connection->SendRemoveComponent(Entry.Key, SpatialConstants::GDK_GAMEPLAY_DEBUGGER_COMPONENT_ID);
+		}
+	}
+
+	TrackedEntities.Empty();
+	ComponentsAdded.Empty();
+	ActorsAdded.Empty();
+	bNeedToUpdateInterest = false;
+
+	NetDriver->Sender->UpdatePartitionEntityInterestAndPosition();
+}
+
+TOptional<VirtualWorkerId> USpatialNetDriverGameplayDebuggerContext::GetActorDelegatedWorkerId(const AActor& InActor)
+{
+	check(NetDriver && NetDriver->PackageMap);
+
+	FEntityData* EntityData = nullptr;
+
+	Worker_EntityId EntityId = NetDriver->PackageMap->GetEntityIdFromObject(&InActor);
+	if (EntityId != SpatialConstants::INVALID_ENTITY_ID)
+	{
+		EntityData = TrackedEntities.Find(EntityId);
+	}
+
+	if (EntityData == nullptr)
+	{
+		return {};
+	}
+
+	return EntityData->Component.DelegatedWorkerId;
+}
+
+void USpatialNetDriverGameplayDebuggerContext::AdvanceView()
```

Comment:
From what I can see there is no distinction here between having authority and not having it.
It is possible that you are only interested in the component if you are authoritative over it. The load balancer "should" only ask you about which worker should have auth if you have authority over it. It would not be true in startup scenarios, but I believe that these actors are only created at runtime and never have to figure out where an actor should go if it does not already have auth over it.

So 2 things : 
Either 1) Change the interest query to only have interest when we have auth over the ServerComponentSet, and add both the component and an additional "tag" component, and set your subview to filter on this tag component. So on and ADD or REMOVE, you will only get the actors you are auth on, which will make a lot of things simpler.
Or 2) Add a flag to know if you have authority over the actor or not.

Reply:
As discussed offline, option 1 won't work.

I already had option 2 in place, although not through an explicit flag, but rather through comparing component/workerid with local.. so I won't duplicate this state.